rpcs3/Emu/Cell/PPUThread.cpp

```diff
@@ -424,19 +440,33 @@ std::string ppu_thread::dump() const
 	fmt::append(ret, "XER = [CA=%u | OV=%u | SO=%u | CNT=%u]\n", xer.ca, xer.ov, xer.so, xer.cnt);
 	fmt::append(ret, "VSCR = [SAT=%u | NJ=%u]\n", sat, nj);
 	fmt::append(ret, "FPSCR = [FL=%u | FG=%u | FE=%u | FU=%u]\n", fpscr.fl, fpscr.fg, fpscr.fe, fpscr.fu);
-	fmt::append(ret, "\nCall stack:\n=========\n0x%08x (0x0) called\n", cia);
 
-	//std::shared_lock rlock(vm::g_mutex); // Needs optimizations
+	return ret;
+}
 
-	// Determine stack range
-	u32 stack_ptr = static_cast<u32>(gpr[1]);
+std::string ppu_thread::dump_callstack() const
+{
+	std::string ret;
+
+	fmt::append(ret, "Call stack:\n=========\n0x%08x (0x0) called\n", cia);
 
-	if (!vm::check_addr(stack_ptr, 1, vm::page_writable))
+	for (u32 sp : dump_callstack_list())
 	{
-		// Normally impossible unless the code does not follow ABI rules
-		return ret;
+		// TODO: print also function addresses
+		fmt::append(ret, "> from 0x%04llx (0x0)\n", vm::read64(static_cast<u32>(sp + 16)));
 	}
 
+	return ret;
+}
+
+std::vector<u32> ppu_thread::dump_callstack_list() const
+{
+	std::vector<u32> call_stack_list;
+
+	//std::shared_lock rlock(vm::g_mutex); // Needs optimizations
+
+	// Determine stack range
+	const u32 stack_ptr = static_cast<u32>(gpr[1]);
```

Comment:
r1 may point to invalid memory in some theoretical cases. (e.g. usermade asm) A vm::check_addr was removed from the rebase for it.